app/javascript/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { ErrorInfo, ReactNode } from "react"
+import { ActionBlock, AppearanceStyleType, Button, t } from "@bloom-housing/ui-components"
+import "./ErrorBoundary.scss"
+
+export enum BoundaryScope {
+  /**
+   * The main content of the page will be wrapped in the error boundary, e.g. outside wrapping components like header
+   * and footer should still show on the page
+   */
+  content = "content",
+
+  /**
+   * This is assumed critical. User will be redirected to an html error page.
+   */
+  page = "page",
+}
+
+interface Props {
+  boundaryScope: BoundaryScope
+  children: ReactNode
+}
+
+interface State {
+  hasError: boolean
+}
+
+class ErrorBoundary extends React.Component<Props, State> {
```

Comment:
If possible we should use exclusively functional components

Reply:
I agree. Unfortunately, react currently only supports error boundaries using class components.